catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.java

```diff
@@ -390,6 +390,12 @@ public void handleUploadButton(View view) {
 		if (!viewSwitchLock.tryLock()) {
 			return;
 		}
+
+		if (BuildConfig.CREATE_AT_SCHOOL && !Utils.isCreateAtSchoolUser(this)) {
+			SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this);
+			sharedPreferences.edit().putString(Constants.TOKEN, Constants.NO_TOKEN).apply();
```

Comment:
(I know that this is already like this in other parts of the code)

if you edit multiple items you should call in this order:
```
SharedPreferences.Editor prefEditor = sharedPreferences.edit();
prefEditor.putString();
prefEditor.putString();
//... all other operations on this editor
prefEditor.apply();
```
for the following reasons: 

- apply() is an atomic write operation, meaning no part of the app could access the preferences in a weird state where for example Constants.TOKEN is written, but Constants.USERNAME still has the old value.
- Each time you call edit() a new SharedPreferences.Editor is created which creates a full copy of the original preferences and each apply() writes your modiefied copy back to the original -> performance

please check all your code bits which use this